src/resolvers/deck/createDeck.ts

```diff
@@ -28,3 +35,190 @@ export const createDeck = mutationWithClientMutationId({
     return { deck }
   },
 })
+
+export const importDeck = mutationWithClientMutationId({
+  name: 'ImportDeck',
+  description: 'Import/Duplicate a deck entity',
+  inputFields: {
+    id: { type: GraphQLNonNull(GraphQLID), description: 'Public deck id' },
+  },
+  outputFields: {
+    deck: { type: DeckType, description: 'Imported deck' },
+  },
+  mutateAndGetPayload: async (
+    { id },
+    { user, publishedDeckLoader, modelLoader, fieldsByModelLoader }: Context
+  ) => {
+    const helperMap: {
+      updateModelIndex: { oldId: string; newId: string }[]
+      updateFieldIndex: { oldId: string; newId: string }[]
+    } = {
+      updateModelIndex: [],
+      updateFieldIndex: [],
+    }
+    const { id: deckId } = fromGlobalId(id)
+    const publishedDeck = await publishedDeckLoader.load(deckId)
```

Comment:
can you do an early return here if the published deck is `null`?

Reply:
This should be resolved with like 62

```typescript
  if (!publishedDeck) {
      return
    }
```